cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -215,6 +217,79 @@ extends ColumnStore with CassandraChunkSource with StrictLogging {
     }
   }
 
+  /**
+    * Copy a range of chunks to a target ColumnStore, for performing disaster recovery or
+    * backfills.
+    */
+  // scalastyle:off null method.length
+  def copyChunksByIngestionTimeRange(datasetRef: DatasetRef,
+                                     splits: Iterator[ScanSplit],
+                                     ingestionTimeStart: Long,
+                                     ingestionTimeEnd: Long,
+                                     batchSize: Int,
+                                     target: CassandraColumnStore,
+                                     targetDatasetRef: DatasetRef,
+                                     diskTimeToLiveSeconds: Int): Unit =
+  {
+    val sourceIndexTable = getOrCreateIngestionTimeIndexTable(datasetRef)
+    val sourceChunksTable = getOrCreateChunkTable(datasetRef)
+
+    val targetIndexTable = target.getOrCreateIngestionTimeIndexTable(targetDatasetRef)
+    val targetChunksTable = target.getOrCreateChunkTable(targetDatasetRef)
+
+    val chunkInfos = new ArrayBuffer[ByteBuffer]()
+    val futures = new ArrayBuffer[Future[Response]]()
+
+    def finishBatch(partition: ByteBuffer): Unit = {
+      for (row <- sourceChunksTable.readChunks(partition, chunkInfos).iterator.asScala) {
+        futures += targetChunksTable.writeChunks(partition, row, diskTimeToLiveSeconds)
+      }
+
+      chunkInfos.clear()
+
+      for (f <- futures) {
+        try {
+          Await.result(f, Duration(10, SECONDS))
+        } catch {
+          case e: Exception => {
+            logger.error(s"Async cassandra chunk copy failed", e)
+          }
+        }
+      }
+
+      futures.clear()
+    }
+
+    var lastPartition: ByteBuffer = null
+
+    for (split <- splits) {
+      val tokens = split.asInstanceOf[CassandraTokenRangeSplit].tokens
+      val rows = sourceIndexTable.scanRowsByIngestionTime(tokens, ingestionTimeStart, ingestionTimeEnd)
+      for (row <- rows) {
+        val partition = row.getBytes(0) // partition
+
+        if (!partition.equals(lastPartition)) {
```

Comment:
The idiomatic way of doing this would be to use takeWhile or groupBy which are not straightforward with Iterators. But I have pointed out elsewhere in this PR that using iterator here is not beneficial at all since we are synchronous and not lazy. We may as well use List.

Reply:
Again, need to understand how the driver works before switching to a List.